src/app/modules/bounce/bounce.service.ts

```diff
@@ -101,20 +108,21 @@ export const isValidSnsRequest = async (
 // Writes a log message if all recipients have bounced
 const logCriticalBounce = (
   bounceDoc: IBounceSchema,
-  formId: string,
   notification: IEmailNotification,
 ): void => {
-  if (
-    !bounceDoc.hasAlarmed &&
-    bounceDoc.bounces.every((emailInfo) => emailInfo.hasBounced)
-  ) {
+  if (bounceDoc.bounces.every((emailInfo) => emailInfo.hasBounced)) {
```

Comment:
Maybe we can make a type guard using this conditional, so you don't need to do `notification as IBounceNotification` casting below

```ts
const isBounceNotification = (doc): doc is IBounceNotification => bounceDoc.bounces.every((emailInfo) => emailInfo.hasBounced)
```

Reply:
alternatively, introduce an instance method `bounceDoc.allEmailsBounced()` to the Domain so that the business logic is clearly represented and elevated in importance

Reply:
the problem here is that we are inferring the type of one object based on another object, i.e. we are inferring that the notification must be a bounce notification because the bounce document shows a critical bounce. I solved this by adding an explicit type check for the notification type, but will explore more options in the PR for auto deactivation and auto emailing, e.g. investigating canonical ways to implement DTOs